### PR TITLE
fix: attempt to fix several issues related to the new cache

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request attempts to fixes [this](https://canary.discord.com/channels/789032594456576001/876490879815254097/996840935356895253) bug
## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [x] Bugfix

### CODE REVIEW

**Comment by Toricane** [comment]
Have you tested it?

**Comment by sillyElvy** [comment]
> Have you tested it?

The issue seems to occur randomly so I can't really test it/we don't know when it occurs

**Review by Toricane — DISMISSED** [review]
LGTM!

**Comment by Damego** [comment]
Now this
```py
Traceback (most recent call last):
  File "C:\Users\User\OneDrive\python\discord bots\mangalib-bot\source\main.py", line 23, in <module>
    client.start()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 134, in start
    raise e from e
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 132, in start
    self._loop.run_until_complete(self._ready())
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\Lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 391, in _ready
    await self._login()
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 396, in _login
    await self._websocket._establish_connection(self._shards, self._presence)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 196, in _establish_connection
    await self._handle_connection(stream, shard, presence)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 264, in _handle_connection
    self._dispatch_event(event, data)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 414, in _dispatch_event
    before = model(**old_obj._json)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\models\attrs_utils.py", line 120, in __init__
    super().__init__(**kwargs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\models\attrs_utils.py", line 80, in __init__
    self.__attrs_init__(**passed_kwargs)  # type: ignore
  File "<attrs generated init interactions.api.models.message.Message>", line 8, in __attrs_init__
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\attr\converters.py", line 43, in optional_converter
    return converter(val)
TypeError: fromisoformat: argument must be str
```

**Comment by Damego** [comment]
> LGTM!

nonono LBTM

**Comment by sillyElvy** [comment]
@Damego merge or no merge?

**Comment by Damego** [comment]
> Now this
> 
> ```python
> Traceback (most recent call last):
>   File "C:\Users\User\OneDrive\python\discord bots\mangalib-bot\source\main.py", line 23, in <module>
>     client.start()
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 134, in start
>     raise e from e
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 132, in start
>     self._loop.run_until_complete(self._ready())
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\Lib\asyncio\base_events.py", line 646, in run_until_complete
>     return future.result()
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 391, in _ready
>     await self._login()
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\client\bot.py", line 396, in _login
>     await self._websocket._establish_connection(self._shards, self._presence)
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 196, in _establish_connection
>     await self._handle_connection(stream, shard, presence)
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 264, in _handle_connection
>     self._dispatch_event(event, data)
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\gateway\client.py", line 414, in _dispatch_event
>     before = model(**old_obj._json)
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\models\attrs_utils.py", line 120, in __init__
>     super().__init__(**kwargs)
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\interactions\api\models\attrs_utils.py", line 80, in __init__
>     self.__attrs_init__(**passed_kwargs)  # type: ignore
>   File "<attrs generated init interactions.api.models.message.Message>", line 8, in __attrs_init__
>   File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\attr\converters.py", line 43, in optional_converter
>     return converter(val)
> TypeError: fromisoformat: argument must be str
> ```

@EdVraz

**Comment by sillyElvy** [comment]
ok ill look

**Review by Damego — APPROVED** [review]
Now LGTM!

File: interactions/api/cache.py [hunk 1]
```diff
@@ -61,16 +61,19 @@ def merge(self, item: _T, id: Optional["Key"] = None) -> None:
                 continue
                 # we can only assume that discord did not provide it, falsely deleting is worse than not deleting
             if getattr(old_item, attrib) != getattr(item, attrib):
-                if isinstance(item.attrib, list) and not isinstance(
-                    old_item.attrib, list
+
+                if isinstance(getattr(item, attrib), list) and not isinstance(
+                    getattr(old_item, attrib), list
```
Comment (Damego) [inline interactions/api/cache.py:66]: You can use walrus operator here

Comment (sillyElvy) [reply interactions/api/cache.py:66]: I thought of it but meh

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**